src/react-components/room/useObjectList.js

```diff
@@ -0,0 +1,242 @@
+
+  const selectObject = useCallback(
+    object => {
+      const cameraSystem = scene.systems["hubs-systems"].cameraSystem;
+
+      setSelectedObject(object);
+
+      if (object.el.object3D !== cameraSystem.inspectable) {
+        if (cameraSystem.inspectable) {
+          cameraSystem.uninspect(false);
+        }
+
+        cameraSystem.enableLights = object.enableLights;
+        cameraSystem.inspect(object.el, 1.5, true, false);
+      }
+    },
+    [scene, setSelectedObject]
+  );
+
+  const deselectObject = useCallback(
+    () => {
+      const cameraSystem = scene.systems["hubs-systems"].cameraSystem;
+
+      setSelectedObject(null);
+
+      cameraSystem.enableLights = true;
+      cameraSystem.uninspect(false);
+
+      if (focusedObject) {
+        cameraSystem.enableLights = focusedObject.enableLights;
+        cameraSystem.inspect(focusedObject.el, 1.5, true, false);
+      }
+    },
+    [scene, setSelectedObject, focusedObject]
+  );
+
+  const focusObject = useCallback(
+    object => {
+      const cameraSystem = scene.systems["hubs-systems"].cameraSystem;
+
+      setFocusedObject(object);
+
+      if (object.el.object3D !== cameraSystem.inspectable) {
+        if (cameraSystem.inspectable) {
+          cameraSystem.uninspect(false);
+        }
+
+        cameraSystem.enableLights = object.enableLights;
+        cameraSystem.inspect(object.el, 1.5, true, false);
+      }
+    },
+    [scene, setFocusedObject]
+  );
+
+  const unfocusObject = useCallback(
+    () => {
+      const cameraSystem = scene.systems["hubs-systems"].cameraSystem;
+
+      setFocusedObject(null);
+
+      cameraSystem.enableLights = true;
+      cameraSystem.uninspect(false);
+
+      if (selectedObject) {
+        cameraSystem.enableLights = selectedObject.enableLights;
+        cameraSystem.inspect(selectedObject.el, 1.5, true, false);
+      }
+    },
+    [scene, setFocusedObject, selectedObject]
+  );
+
+  const selectNextObject = useCallback(
+    () => {
+      const curObjIdx = objects.indexOf(selectedObject);
+
+      if (curObjIdx !== -1) {
+        const nextObjIdx = (curObjIdx + 1) % objects.length;
+        selectObject(objects[nextObjIdx]);
+      }
+    },
+    [selectObject, objects, selectedObject]
+  );
+
+  const selectPrevObject = useCallback(
+    () => {
+      const curObjIdx = objects.indexOf(selectedObject);
+
+      if (curObjIdx !== -1) {
+        const nextObjIdx = curObjIdx === 0 ? objects.length - 1 : curObjIdx - 1;
+        selectObject(objects[nextObjIdx]);
+      }
+    },
+    [selectObject, objects, selectedObject]
+  );
+
+  const context = {
+    objects,
+    activeObject: focusedObject || selectedObject,
+    focusedObject,
+    selectedObject,
+    focusObject,
+    unfocusObject,
+    selectObject,
+    deselectObject,
+    selectPrevObject,
+    selectNextObject
+  };
+
+  // Note: If we move ui-root to a functional component and use hooks,
+  // we can use the useObjectList hook instead of cloneElement.
```

Comment:
Should this remain in the code, or is it a personal TODO for later?

Reply:
Yeah, I'm working towards this, but it could take a while and may even happen after we ship the redesign. I'd rather keep this in code so people know why I'm using `cloneElement` here in case they run across it before things change.